### PR TITLE
fix!: upgrade node to v20, fix abort handling

### PR DESCRIPTION
**Pull request by nordfjord**

### CODE REVIEW

File: packages/propeller/src/Queue.ts [hunk 1]
```diff
@@ -88,7 +88,7 @@ export class AsyncQueue<T> {
     const value = this.queue.tryFind(predicate)
     if (value) return value
     return new Promise<T>((resolve, reject) => {
-      const abort = () => reject(new Error("Aborted"))
+      const abort = () => reject(new DOMException("The operation was aborted", "AbortError"))
```
Comment (sroze) [inline packages/propeller/src/Queue.ts:91]: @nordfjord why `DOMException` in this very backend-y package? It means it doesn't work with Node < 20, either.

Comment (sroze) [reply packages/propeller/src/Queue.ts:91]: image

Comment (nordfjord) [reply packages/propeller/src/Queue.ts:91]: Because that's what node throws from `AbortController#abort`

Comment (nordfjord) [reply packages/propeller/src/Queue.ts:91]: Would be even worse to have multiple types of exceptions representing abort errors IMO

Comment (nordfjord) [reply packages/propeller/src/Queue.ts:91]: See https://github.com/nodejs/node/blob/main/lib/internal/abort_controller.js#L178-L179

**Pull request merged**